Mathlib/Topology/UniformSpace/Dini.lean

```diff
@@ -0,0 +1,152 @@
+/-
+Copyright (c) 2024 Jireh Loreaux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jireh Loreaux
+-/
+import Mathlib.Analysis.Normed.Order.Lattice
+import Mathlib.Topology.ContinuousMap.Ordered
+import Mathlib.Topology.UniformSpace.CompactConvergence
+
+/-! # Dini's Theorem
+
+This file proves Dini's theorem, which states that if `F n` is a monotone increasing sequence of
+continuous real-valued functions on a compact set `s` converging pointwise to a continuous function
+`f`, then `F n` converges uniformly to `f`.
+
+We generalize the codomain from `ℝ` to a normed lattice additive commutative group `G`.
+This theorem is true in a different generality as well: when `G` is a linearly ordered topological
+group with the order topology. This weakens the norm assumption, in exchange for strengthening to
+a linear order. This separate generality is not included in this file, but that generality was
+included in initial drafts of the original
+[PR #19068](https://github.com/leanprover-community/mathlib4/pull/19068) and can be recovered if
+necessary.
```

Comment:
I'm completely ready to believe there is a good generalization, but it's not obvious to me, probably because I don't really know what are the right analogues of [OrderTopology](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Order/Basic.html#OrderTopology) in the non-linearly-ordered setting (I know for sure that working with open intervals is bad, and I think the version with closed intervals still behaves badly sometimes, e.g it's still not stable under products unless there are top and bottom elements).

What we really need in our proof is a basis of neighborhoods of zero which is stable under having smaller absolute value. We achieve this in our setting thanks to [norm_le_norm_of_abs_le_abs](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Analysis/Normed/Order/Lattice.html#norm_le_norm_of_abs_le_abs), and in the order topology case it's obvious. What is the precise generality you have in mind ?

Reply:
I see. Can you turn a combination of both your answers into an extra comment in the module doc?